### PR TITLE
Fixes eclipse/lsp4j#71: ignore unexpected params

### PR DESCRIPTION
**Pull request by akosyakov**

### CODE REVIEW

File: org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/services/GenericEndpoint.java [hunk 1]
```diff
@@ -28,6 +29,7 @@
 public class GenericEndpoint implements Endpoint {
 	
 	private static final Logger LOG = Logger.getLogger(GenericEndpoint.class.getName());
+	private static Object[] NO_ARGUMENTS = {};
```
Comment (kittaakos) [inline org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/services/GenericEndpoint.java]: It is a minor thing, and `NO_ARGUMENTS` an array anyway, but still, could we make this final?

Comment (akosyakov) [inline org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/services/GenericEndpoint.java]: @kittaakos thx!

File: org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/services/GenericEndpoint.java [hunk 2]
```diff
@@ -42,8 +44,9 @@ protected void recursiveFindRpcMethods(Object current, Set<Class<?>> visited, Se
 			@SuppressWarnings("unchecked")
 			Function<Object, CompletableFuture<Object>> handler = (arg) -> {
 				try {
-					Object[] argument = arg == null ? new Object[0] : new Object[] { arg };
-					return (CompletableFuture<Object>) methodInfo.method.invoke(current, argument);
+					Method method = methodInfo.method;
+					Object[] arguments = this.getArguments(method, arg);
```
Comment (kittaakos) [inline org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/services/GenericEndpoint.java]: Too much TS code, right? :) You could omit `this` just to align your changes with the existing code style.

File: org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/services/GenericEndpoint.java [hunk 3]
```diff
@@ -62,6 +65,17 @@ protected void recursiveFindRpcMethods(Object current, Set<Class<?>> visited, Se
 			}
 		});
 	}
+	
+	protected Object[] getArguments(Method method, Object arg) {
+		if (arg == null) {
+			return NO_ARGUMENTS;
+		}
+		if (method.getParameterCount() == 0) {
+			LOG.warning("Unexpected params '" + arg + "' for '" + method + "' is ignored");
```
Comment (kittaakos) [inline org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/services/GenericEndpoint.java]: Yet another small comment: here, maybe we could distinguish between single objects and Arrays to have a better log message.

Comment (akosyakov) [inline org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/services/GenericEndpoint.java]: actually the value does not matter, the point that the argument should not passed at all

**Pull request merged**